tests/manager/api/test_app.py

```diff
@@ -10,9 +10,14 @@ def test_initialize_application_http(
     db: DatabaseTransactionFixture, services_fixture_wired: ServicesFixture
 ):
     # Use the db transaction fixture so that we don't use the production settings by mistake
-    with patch.object(
-        HTTP, "set_quick_failure_settings"
-    ) as mock_set_quick_failure_settings:
+    with (
+        patch.object(
+            HTTP, "set_quick_failure_settings"
+        ) as mock_set_quick_failure_settings,
+        patch("palace.manager.api.app.initialize_database"),
+        patch("palace.manager.api.app.initialize_circulation_manager"),
+        patch("palace.manager.api.app.app"),
+    ):
```

Comment:
We need some mocks here, otherwise this test trips over the patch we apply that raises an exception if a test tries to create an engine directly. The test wasn't very safe before because it was doing a full app initialization, but leaking the database connection.